extensions/arc/src/models/postgresModel.ts

```diff
@@ -0,0 +1,152 @@
+	}
+
+	/** Returns the number of nodes in the service */
+	public numNodes(): number {
+		let nodes = this._service.spec.scale?.shards ?? 1;
+		if (nodes > 1) { nodes++; } // for multiple shards there is an additional node for the coordinator
+		return nodes;
+	}
+
+	/**
+	 * Returns the IP address and port of the service, preferring external IP over
+	 * internal IP. If either field is not available it will be set to undefined.
+	 */
+	public endpoint(): { ip?: string, port?: number } {
+		const externalIp = this._service.status?.externalIP;
+		const internalIp = this._service.status?.internalIP;
+		const externalPort = this._service.status?.externalPort;
+		const internalPort = this._service.status?.internalPort;
+
+		return externalIp ? { ip: externalIp, port: externalPort ?? undefined }
+			: internalIp ? { ip: internalIp, port: internalPort ?? undefined }
+				: { ip: undefined, port: undefined };
+	}
+
+	/** Returns the service's configuration e.g. '3 nodes, 1.5 vCores, 1GiB RAM, 2GiB storage per node' */
+	public configuration(): string {
+		const nodes = this.numNodes();
+		const cpuLimit = this._service.spec.scheduling?.resources?.limits?.['cpu'];
+		const ramLimit = this._service.spec.scheduling?.resources?.limits?.['memory'];
+		const cpuRequest = this._service.spec.scheduling?.resources?.requests?.['cpu'];
+		const ramRequest = this._service.spec.scheduling?.resources?.requests?.['memory'];
+		const storage = this._service.spec.storage.volumeSize;
+
+		// Prefer limits if they're provided, otherwise use requests if they're provided
+		let nodeConfiguration = `${nodes} node`;
+		if (nodes > 1) { nodeConfiguration += 's'; }
+		if (cpuLimit) {
+			nodeConfiguration += `, ${this.formatCores(cpuLimit)} vCores`;
+		} else if (cpuRequest) {
+			nodeConfiguration += `, ${this.formatCores(cpuRequest)} vCores`;
+		}
+		if (ramLimit) {
+			nodeConfiguration += `, ${this.formatMemory(ramLimit)} RAM`;
+		} else if (ramRequest) {
+			nodeConfiguration += `, ${this.formatMemory(ramRequest)} RAM`;
+		}
+		if (storage) { nodeConfiguration += `, ${storage} storage per node`; }
+		return nodeConfiguration;
+	}
+
+	/**
+	 * Converts millicores to cores (600m -> 0.6 cores)
+	 * https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-cpu
+	 * @param cores The millicores to format e.g. 600m
+	 */
+	private formatCores(cores: string): number {
+		return cores?.endsWith('m') ? +cores.slice(0, -1) / 1000 : +cores;
+	}
+
+	/**
+	 * Formats the memory to end with 'B' e.g:
+	 * 1 -> 1B
+	 * 1K -> 1KB, 1Ki -> 1KiB
+	 * 1M -> 1MB, 1Mi -> 1MiB
+	 * 1G -> 1GB, 1Gi -> 1GiB
+	 * 1T -> 1TB, 1Ti -> 1TiB
+	 * https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#meaning-of-memory
+	 * @param memory The amount + unit of memory to format e.g. 1K
+	 */
+	private formatMemory(memory: string): string {
+		return memory && !memory.endsWith('B') ? `${memory}B` : memory;
```

Comment:
This should be localized

Reply:
Interesting, how should an amount of memory like '1GB' be localized?  It comes from k8s as '1G' and this function adds the 'B'.

Reply:
I assume this means the unit (B, KB, KiB, etc) should be localized.  I can add localized constants for those.

Reply:
Well that's weird. Ok for now let's just leave these as is until we decide it's worth spending time looking into how best to handle this.